lua/luasnip/extras/postfix.lua

```diff
@@ -2,53 +2,61 @@ local snip = require("luasnip.nodes.snippet").S
 local events = require("luasnip.util.events")
 local extend_decorator = require("luasnip.util.extend_decorator")
 local node_util = require("luasnip.nodes.util")
+local util = require("luasnip.util.util")
 
 local matches = {
 	default = [[[%w%.%_%-%"%']+$]],
 	line = "^.+$",
 }
 
-local function generate_opts(match_pattern, user_callback)
+local function wrap_resolve_expand_params(match_pattern, user_resolve)
+	return function(snippet, line_to_cursor, match, captures)
+		if snippet.condition then
+			if not snippet.condition(line_to_cursor, match, captures) then
+				return nil
+			end
+		end
+
+		local pos = util.get_cursor_0ind()
+		local line_to_cursor_except_match =
+			line_to_cursor:sub(1, #line_to_cursor - #match)
+		local postfix_match = line_to_cursor_except_match:match(match_pattern)
+			or ""
+		local res = {
+			trigger = match,
+			captures = captures,
+			clear_region = {
+				from = { pos[1], pos[2] - #postfix_match - #match },
+				to = pos,
+			},
+			env_override = {
+				POSTFIX_MATCH = postfix_match,
+			},
+		}
+
+		if user_resolve then
+			local user_res =
+				user_resolve(snippet, line_to_cursor, match, captures)
+			if user_res then
+				res = vim.tbl_extend("force", res, user_res)
+			else
+				return nil
+			end
+		end
+		return res
+	end
+end
+
+local function generate_opts(user_callback)
 	return {
 		callbacks = {
 			[-1] = {
 				[events.pre_expand] = function(snippet, event_args)
-					local pos = event_args.expand_pos
-					-- [1]: returns table, gets text end-exclusive.
-					local line_to_cursor = vim.api.nvim_buf_get_text(
-						0,
-						pos[1],
-						0,
-						pos[1],
-						pos[2],
-						{}
-					)[1]
-					local postfix_match = line_to_cursor:match(match_pattern)
-						or ""
-					-- clear postfix_match-text.
-					vim.api.nvim_buf_set_text(
-						0,
-						pos[1],
-						pos[2] - #postfix_match,
-						pos[1],
-						pos[2],
-						{ "" }
-					)
 					local user_env = {}
 					if user_callback then
 						user_env = user_callback(snippet, event_args) or {}
 					end
-					local postfix_env_override = {
-						env_override = {
-							POSTFIX_MATCH = postfix_match,
-						},
-					}
-
-					return vim.tbl_deep_extend(
-						"keep",
-						user_env,
-						postfix_env_override
-					)
+					return user_env
 				end,
```

Comment:
I think this can be removed entirely, all it does now is call the user-callback, and this also works by just passing `opts` to `snip(..)`

Reply:
you are right. i will fix it later.